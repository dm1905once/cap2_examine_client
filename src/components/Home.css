/* Banner */

#banner {
    padding: 8em 0 9em 0;
    background-image: url(../images/nathan-dumlao-ewGMqs2tmJI-unsplash.jpg);
    background-size: cover;
    background-position: bottom;
    background-attachment: fixed;
    background-repeat: no-repeat;
    text-align: center;
    position: relative;
}

    #banner:before {
        content: '';
        background: rgba(173, 216, 230, 0.7);
        position: absolute;
        width: 100%;
        height: 100%;
        top: 0;
        left: 0;
    }

    #banner .inner {
        /* border-top: 2px solid rgba(255, 255, 255, 0.2); */
        position: relative;
        z-index: 10005;
        padding-top: 8em;
    }

    #banner h1 {
        font-size: 5em;
        font-weight: 600;
        color: #000050;
        line-height: 2em;
        margin: 0 0 1em 0;
        padding: 0;
    }

    #banner h3 {
        font-weight: 400;
        color: #000050;
        margin: 0;
        font-size: 2em;
    }

    #banner .icon {
        color: #000050;
        font-size: 5em;
    }

    #banner p {
        font-size: 1.25em;
        color: rgba(0, 0, 0, 0.9);
        margin-bottom: 1.75em;
    }

    #banner .flex {
        -ms-flex-pack: center;
        -moz-justify-content: center;
        -webkit-justify-content: center;
        -ms-justify-content: center;
        justify-content: center;
        text-align: center;
        margin: 0 auto 4em auto;
    }

        #banner .flex div {
            border-right: 2px solid rgba(255, 255, 255, 0.2);
            padding: 0 8em;
        }

            #banner .flex div:last-child {
                border: none;
                padding-right: 0;
            }

            #banner .flex div:first-child {
                padding-left: 0;
            }

            #banner .flex div p {
                margin: 0;
            }



.image {
    border-radius: 0;
    border: 0;
    display: inline-block;
    position: relative;
}

    .image.round img {
        border-radius: 100%;
    }


	section.wrapper, article.wrapper {
		padding: 6em 0;
	}

		section.wrapper header, article.wrapper header {
			margin-bottom: 4em;
		}

			section.wrapper header h2, article.wrapper header h2 {
				font-size: 2.75em;
				margin: 0 0 .5em 0;
			}

			section.wrapper header p, article.wrapper header p {
				font-size: 1em;
			}

		section.wrapper article header, article.wrapper article header {
			margin: 0;
        }
        

        .flex {
            display: -ms-flexbox;
            -ms-flex-wrap: wrap;
            -ms-flex-pack: justify;
            -moz-justify-content: space-between;
            -webkit-justify-content: space-between;
            -ms-justify-content: space-between;
            justify-content: space-between;
            display: -moz-flex;
            display: -webkit-flex;
            display: -ms-flex;
            display: flex;
            -moz-flex-wrap: wrap;
            -webkit-flex-wrap: wrap;
            -ms-flex-wrap: wrap;
            flex-wrap: wrap;
            width: 100%;
        }
    
            .flex.flex-2 {
                -moz-justify-content: space-between;
                -webkit-justify-content: space-between;
                -ms-justify-content: space-between;
                justify-content: space-between;
            }
    
                .flex.flex-2 article {
                    width: 50%;
                }


hr {
    border: 0;
    border-bottom: solid 1px #dbdbdb;
    margin: 2em 0;
}

    hr.major {
        margin: 3em 0;
    }

.align-left {
    text-align: left;
}

.align-center {
    text-align: center;
}

.align-right {
    text-align: right;
}

.inner {
        max-width: 75em;
        margin: 0 auto;
    }


    h1, h2, h3, h4, h5, h6 {
		color: #555;
		font-weight: 400;
		line-height: 1.5;
		margin: 0 0 1em 0;
	}

		h1 a, h2 a, h3 a, h4 a, h5 a, h6 a {
			color: inherit;
			text-decoration: none;
		}

	h2 {
		font-size: 1.85em;
		font-weight: 300;
	}

	h3 {
		font-size: 1.75em;
    }
    
a.access-link {
    color: #000050;
    font-size: 2em;
    font-weight: 400;
    line-height: 2.5;
    margin: 0 0 1em 0;
}

    a.access-link:hover {
        color: lightblue;
    }


    #footer {
		padding: 0.25em 0;
		background: rgba(173, 216, 230, 0.7);
		color: black;
		text-align: center;
	}

    

