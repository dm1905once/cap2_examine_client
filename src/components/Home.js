import React from 'react';
import { Link } from 'react-router-dom';
import './Home.css';
import image_app_access from '../images/tim-gouw-ScWvHUtQca4-unsplash_190x190.jpg';
import image_org_access from '../images/element5-digital-EzYCojqgbgM-unsplash_190x190.jpg';

const Home = ()=> {
    return (
        <div>
            <section id="banner">
				<div className="inner">
					<header>
						<h1>Welcome to Examine</h1>
					</header>

					<div className="flex">

						<div>
							<i className="edit outline icon"></i>
							<h3>The convenient online evaluation platform</h3>
							<p>Educational organizations take advantage of the Examine platform to securely apply exams to thousands of candidates</p>
						</div>

					</div>

				</div>
			</section>


			<section id="three" className="wrapper align-center">
				<div className="inner">
					<div className="flex flex-2">
						<article>
							<div className="image round">
								<img src={image_app_access} alt="Applicant access" />
							</div>
							<header>
                                <Link className="access-link" to='/applicants'>Access as an Applicant</Link>
							</header>
							<p>Applicants are exam takers. Once they register for free they can browse, select and purchase the right to take an exam.</p>
							<footer></footer>
						</article>
						<article>
							<div className="image round">
								<img src={image_org_access} alt="Org access" />
							</div>
							<header>
                                <Link className="access-link" to='orgs'>Access as Educational Organization</Link>
							</header>
							<p>Educational Organizations are schools, colleges, universities or businesses that require applicants to take an exam to be admitted.</p>
							<footer></footer>
						</article>
					</div>
				</div>
			</section>

            <footer id="footer">
				<div className="inner">Photos by:
                <p><a href="https://unsplash.com/@nate_dumlao?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Nathan Dumlao</a> on Unsplash</p>
                <p><a href="https://unsplash.com/@punttim?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Tim Gouw</a> on Unsplash</p>
                <p><a href="https://unsplash.com/@element5digital?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Element5 Digital</a> on Unsplash</p>
                </div>
            </footer>


        </div>
    )
}

export default Home;