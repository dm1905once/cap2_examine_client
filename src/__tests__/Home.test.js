import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/Home';

// Smoke test
test('Home renders without crashing', ()=>{
    render(<BrowserRouter><Home /></BrowserRouter>);
});


// Snapshot test
test('Home matches snapshot', ()=>{
    const { asFragment } = render(<BrowserRouter><Home /></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
})