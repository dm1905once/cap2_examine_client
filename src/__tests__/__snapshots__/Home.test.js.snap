// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <section
      id="banner"
    >
      <div
        class="inner"
      >
        <header>
          <h1>
            Welcome to Examine
          </h1>
        </header>
        <div
          class="flex"
        >
          <div>
            <i
              class="edit outline icon"
            />
            <h3>
              The convenient online evaluation platform
            </h3>
            <p>
              Educational organizations take advantage of the Examine platform to securely apply exams to thousands of candidates
            </p>
          </div>
        </div>
      </div>
    </section>
    <section
      class="wrapper align-center"
      id="three"
    >
      <div
        class="inner"
      >
        <div
          class="flex flex-2"
        >
          <article>
            <div
              class="image round"
            >
              <img
                alt="Applicant access"
                src="tim-gouw-ScWvHUtQca4-unsplash_190x190.jpg"
              />
            </div>
            <header>
              <a
                class="access-link"
                href="/applicants"
              >
                Access as an Applicant
              </a>
            </header>
            <p>
              Applicants are exam takers. Once they register for free they can browse, select and purchase the right to take an exam.
            </p>
            <footer />
          </article>
          <article>
            <div
              class="image round"
            >
              <img
                alt="Org access"
                src="element5-digital-EzYCojqgbgM-unsplash_190x190.jpg"
              />
            </div>
            <header>
              <a
                class="access-link"
                href="/orgs"
              >
                Access as Educational Organization
              </a>
            </header>
            <p>
              Educational Organizations are schools, colleges, universities or businesses that require applicants to take an exam to be admitted.
            </p>
            <footer />
          </article>
        </div>
      </div>
    </section>
    <footer
      id="footer"
    >
      <div
        class="inner"
      >
        <strong>
          Photos by: 
        </strong>
        <span>
          <a
            href="https://unsplash.com/@nate_dumlao?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
          >
            Nathan Dumlao
          </a>
           on Unsplash, 
        </span>
        <span>
          <a
            href="https://unsplash.com/@punttim?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
          >
            Tim Gouw
          </a>
           on Unsplash, 
        </span>
        <span>
          <a
            href="https://unsplash.com/@element5digital?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
          >
            Element5 Digital
          </a>
           on Unsplash, 
        </span>
        <span>
          <a
            href="https://unsplash.com/@jessbaileydesigns?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
          >
            Jess Bailey
          </a>
           on Unsplash
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
